doc/develop-plugin-csharp.md

```diff
@@ -0,0 +1,103 @@
+# Writing a plugin in C#
+
+This document describes how to write a plugin with C#. The C# plugins can only be used in the C# application project. If you want to create a C++ based application, you have to create a plugin using C++. Please see [Writing a new plugin to use platform features](develop-plugin.md) for more information on creating Tizen plugins using C++ and Dart.
+
+
+## Create a C# plugin package
+
+To create a C# plugin package, you need to follow the steps below:
+```sh
+flutter-tizen create --platforms tizen --template plugin --tizen-language csharp [plugin_name]
+```
+Once the package is created, you will be prompted to add some information to its `pubspec.yaml` file. Open the `pubspec.yaml` file in the directory `[plugin_name]` created by the command above and replace the `some_platform:` map with `tizen:` as suggested by the tool. This information is needed by the flutter-tizen tool to find and register the plugin when building an app that depends on the plugin.
+
+```yaml
+You've created a plugin project that doesn't yet support any platforms.
+...
+Make sure your plugin_name/pubspec.yaml contains the following lines.
+
+flutter:
+  plugin:
+    platforms:
+      tizen:
+        namespace: PluginName
+        pluginClass: PluginNamePlugin
+        fileName: PluginNamePlugin.csproj
+```
+
+The created package contains an example app in the `example/` directory. You can run using the following command:
+```sh
+cd plugin_name/example
+flutter-tizen run
+```
+
+## Implement the plugin
+
+### 1. Define the package API (`.dart`)
+
+The API of the plugin package is defined in Dart code. Open `lib/plugin_name.dart` file, and then you will see the `platformVersion` property defined in the plugin main class. Invoking this property will invoke the `getPlatformVersion` method through a method channel named `plugin_name`. You have to replace this template code with the actual implementation for your plugin.
+
+Note: This file is not necessary if you're extending an existing plugin for Tizen, so you can safely remove it.
+
+### 2. Add Tizen platform code (`.cs`)
+
+The implementation of the plugin can be found in the `tizen/PluginNamePlugin.cs` file. This file has a `public` class named `PluginNamePlugin` that implements the `IFlutterPlugin` interface. The `IFlutterPlugin` interface is used to handle the lifecycle of the plugin.
+
+- `OnAttachedToEngine(IFlutterPluginBinding binding)`: This method is called when the plugin is registered to the `DotnetPluginRegistry` by the flutter application. In this method, you can create and initialize the platform channel.
+
+```c#
+        public void OnAttachedToEngine(IFlutterPluginBinding binding)
+        {
+            _channel = new MethodChannel("plugin_name", StandardMethodCodec.Instance, binding.BinaryMessenger);
+            _channel.SetMethodCallHandler(HandleMethodCall);
+        }
+```
```

Comment:
Please indent properly.
```suggestion
  ```c#
  public void OnAttachedToEngine(IFlutterPluginBinding binding)
  {
      _channel = new MethodChannel("plugin_name", StandardMethodCodec.Instance, binding.BinaryMessenger);
      _channel.SetMethodCallHandler(HandleMethodCall);
  }
  ```
```

Reply:
I mean, the entire code block should be indented with extra two spaces because it's part of the `<ul>` element.

Reply:
oh.. I understand now :)